src/BuildScriptGenerator/DefaultBuildScriptGenerator.cs

```diff
@@ -208,16 +216,24 @@ private static string GetBenvArgs(IDictionary<string, string> benvArgsMap)
             IDictionary<string, string> tools,
             [NotNull] List<ICheckerMessage> checkerMessageSink)
         {
-            var checkers = _checkers.WhereApplicable(tools);
+            var checkers = _checkers.WhereApplicable(tools).ToArray();
```

Comment:
all of these parameters should be `[NotNull]`

Reply:
correct, but i wanted to be over-explicit and avoid confusion with `IBuildScriptGenerator.GenerateBashScript` which can receive a null checkerMessageSink.